AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Sync/InitialSync/SyncEventEmitterTests.swift

```diff
@@ -0,0 +1,192 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import XCTest
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSPluginsCore
+@testable import AWSDataStoreCategoryPlugin
+
+class SyncEventEmitterTests: XCTestCase {
+    var initialSyncOrchestrator: MockAWSInitialSyncOrchestrator?
+    var reconciliationQueue: MockAWSIncomingEventReconciliationQueue?
+    var syncEventEmitter: SyncEventEmitter?
+    var expectedModelSyncedEventPayloads: [ModelSyncedEvent]!
+
+    override func setUp() {
+        super.setUp()
+
+        ModelRegistry.reset()
+        MockModelReconciliationQueue.reset()
+        MockAWSInitialSyncOrchestrator.reset()
+
+        expectedModelSyncedEventPayloads = [ModelSyncedEvent(modelName: "Post",
+                                                             isFullSync: true, isDeltaSync: false,
+                                                             added: 1, updated: 0, deleted: 0),
+                                            ModelSyncedEvent(modelName: "Comment",
+                                                             isFullSync: true, isDeltaSync: false,
+                                                             added: 1, updated: 0, deleted: 0)]
```

Comment:
you can define these in-line of the property declaration since they never change

Reply:
if these are always the expected payload, then you can also have corresponding Post and Comments outside of the test itself, as an instance or static variable, and instantatied in the setUp() or in-line of the declaration